frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx

```diff
@@ -0,0 +1,433 @@
+                />
+                <HelpBlock>{validationMsg || null}</HelpBlock>
+              </React.Fragment>
+            );
+            break;
+          case ELEMENT_TYPES.TEXT:
+          default:
+            children = (
+              <React.Fragment>
+                <label
+                  className={classNames('control-label', {
+                    'co-required': parameter.required,
+                  })}
+                >
+                  {_.get(parameter, 'name', key)}
+                </label>
+                <FormControl
+                  type="text"
+                  bsClass="pf-c-form-control"
+                  value={_.get(typeParamsData, `${key}.value`, '')}
+                  onChange={(event) => handleTypeParamChange(key, event, ELEMENT_TYPES.TEXT)}
+                />
+                <HelpBlock>{validationMsg || null}</HelpBlock>
+              </React.Fragment>
+            );
+        }
+
+        return (
+          <FormGroup
+            key={key}
+            controlId={`network-type-parameters-${key}`}
+            validationState={_.get(typeParamsData, `${key}.validationMsg`, null) ? 'error' : null}
+          >
+            {children}
+          </FormGroup>
+        );
+      });
+
+      return (
+        <React.Fragment>
+          {dynamicContent}
+          <div>L2 is the default if nothing else is provided.</div>
+        </React.Fragment>
+      );
+    };
+
+    return getTypeElements();
+  }, [networkType, typeParamsData, setTypeParamsData, sriovNetNodePoliciesData]);
+
+  const formIsValid = React.useCallback(() => {
+    const allTypeParamFieldsValid = () => {
+      return !_.some(typeParamsData, ({ validationMsg }) => validationMsg !== null);
+    };
+
+    const allRequiredFieldsFilled = (): boolean => {
+      if (_.isEmpty(name) || networkType === null) {
+        return false;
+      }
+
+      const allParamsForType = _.get(networkTypeParams, [networkType]);
+      const requiredKeys = _.keys(allParamsForType).filter((key) =>
+        _.get(allParamsForType, [key, 'required'], false),
+      );
+
+      return _.every(requiredKeys, (key) => {
+        const value = _.get(typeParamsData, [key, 'value']);
+        return !_.isEmpty(value);
+      });
+    };
+
+    const validateForm = () => {
+      // Clear error messages from previous validation attempts first
+      const nameIsValid = _.get(fieldErrors, 'nameValidationMsg', '') === '';
+
+      return nameIsValid && allRequiredFieldsFilled() && allTypeParamFieldsValid();
+    };
+
+    return validateForm();
+  }, [fieldErrors, name, networkType, typeParamsData]);
+
+  const getNetworkTypes = React.useCallback(() => {
+    const types = _.clone(networkTypes);
+    if (!hasSriovNetNodePolicyCRD) {
+      delete types.sriov;
+    }
+
+    return types;
+  }, [hasSriovNetNodePolicyCRD]);
+
+  const buildConfig = (): NetworkAttachmentDefinitionConfig => {
```

Comment:
it would be better to move these functions outside of component to not recreate them on every refresh. Applies for the ones below too

Reply:
Done!